metrics/cli/test/cli/updater/updater_test.dart

```diff
@@ -0,0 +1,305 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:cli/cli/updater/algorithm/update_algorithm.dart';
+import 'package:cli/cli/updater/strings/update_strings.dart';
+import 'package:cli/cli/updater/updater.dart';
+import 'package:cli/common/constants/deploy_constants.dart';
+import 'package:cli/common/model/config/update_config.dart';
+import 'package:cli/common/model/paths/factory/paths_factory.dart';
+import 'package:cli/common/model/paths/paths.dart';
+import 'package:cli/common/strings/common_strings.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../test_utils/extension/error_answer.dart';
+import '../../test_utils/matchers.dart';
+import '../../test_utils/mock/directory_mock.dart';
+import '../../test_utils/mock/file_helper_mock.dart';
+import '../../test_utils/mock/path_factory_mock.dart';
+import '../../test_utils/mock/prompter_mock.dart';
+
+// ignore_for_file: avoid_redundant_argument_values, avoid_implementing_value_types, must_be_immutable
+
+void main() {
+  group("Updater", () {
+    const tempDirectoryPath = 'tempDirectoryPath';
+    const prefix = DeployConstants.tempDirectoryPrefix;
+
+    final stateError = StateError('stateError');
+    final paths = Paths(tempDirectoryPath);
+    final updateAlgorithm = _UpdateAlgorithmMock();
+    final fileHelper = FileHelperMock();
+    final prompter = PrompterMock();
+    final pathsFactory = PathsFactory();
```

Comment:
✅ 
It seems like we can use the `pathsFactoryMock` only within the below test cases.